> > AFAIK nixf-diagnose works by _reporting_ issues, not writing / change the files.
> 
> Yes, other tools do the same. `actionlint` doesn't fix anything, but only reports. `editorconfig-checker` currently as well.
> 
> > There might be some issue cannot be fixed trivially by `nixf-diagnose`, how does treefmt catch that?
> 
> It just reports the error. Example from what I had earlier today:
> 
> ```
> [...]
> INFO formatter | editorconfig-checker[1]: 1024 file(s) processed in 3.202169824s
> INFO formatter | nixfmt: 1024 file(s) processed in 16.866898201s
> INFO formatter | editorconfig-checker[1]: 1024 file(s) processed in 3.362892961s
> ERRO formatter | actionlint: failed to apply with options '[]': exit status 1
> 
> .github/workflows/eval.yml:200:20: property "get-merge-commit" is not defined in object type {process: {outputs: {targetrunid: string}; result: string}} [expression]
>     |
> 200 |           ref: ${{ needs.get-merge-commit.outputs.targetSha }}
>     |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> .github/workflows/get-merge-commit.yml:36:9: shellcheck reported issue in this script: SC2028:info:7:12: echo may not expand escape sequences. Use printf [shellcheck]
>    |
> 36 |         run: |
>    |         ^~~~
> 
> INFO formatter | keep-sorted: 18 file(s) processed in 9.616423ms
> INFO formatter | editorconfig-checker[1]: 1024 file(s) processed in 3.353807241s
> INFO formatter | editorconfig-checker[1]: 1024 file(s) processed in 3.198558746s
> [...]
> ```
> 
> That's the actionlint output inbetween.


Did you mean that I need to modify `nixf-diagnose`, and accept CLI flags to conform [formatters-spec](https://treefmt.com/formatters-spec/)?

--- a/pkgs/by-name/ni/nixf-diagnose/package.nix
+++ b/pkgs/by-name/ni/nixf-diagnose/package.nix
@@ -1,0 +1,31 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  nixf,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "nixf-diagnose";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "inclyc";
+    repo = "nixf-diagnose";
+    tag = finalAttrs.version;
+    hash = "sha256-gkeU3EwAl9810eRRp5/ddf1h0qpV6FrBBdntNBpBtsM=";
+  };
+
+  env.NIXF_TIDY_PATH = lib.getExe nixf;
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-nrr2/lTWPyH7MsG2hSMJjbFCpHsKWINEP8jwSYPhocg=";
+
+  meta = {
+    description = "CLI wrapper for nixf-tidy with fancy diagnostic output";
+    mainProgram = "nixf-diagnose";
+    homepage = "https://github.com/inclyc/nixf-diagnose";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ inclyc ];
+  };
+})

